File: Mu2e-STMDAQ/OLD_BUT_DONT_DELETE_YET/NOT_NEEDED/hardwareScripts/FMC120/FMC120_freqcnt.cc
/////////////////////////////////////////////////////////////////////
/// This module obtains various frequencies from the clock tree
/////////////////////////////////////////////////////////////////////

#include <unistd.h>

// FMC120_freqcnt
#include "STMDAQ-TestBeam/hardwareScripts/FMC120/FMC120_freqcnt.hh"

//Standard constructor - shouldn't be used
FMC120_freqcnt::FMC120_freqcnt() {}

// Obtain a given frequency from the clock tree
// This function can either display the frequency on the console or not. 
int FMC120_freqcnt::getfrequency(IPBusManager* hw, uint32_t clksel){

  // tell the firmware to start a measure on a given clock index
  hw->write("axi_fmc120_8lane.Freq_ctr.clk_sel",clksel);
  sleep(0.5);
  
  // read back the just measured value
  uint32_t dword = hw->read("axi_fmc120_8lane.Freq_ctr.clk_cnt");
  double cmdfreq = 125.0;
  double testClkperiod = 1.0 / 125;
  double tmp = 8192 * testClkperiod;
  tmp = (tmp / (dword + 1));
  tmp = 1.00 / tmp;
  int mult_factor = 4;
  if (clksel == 0) {
    std::cout << "Stellar IP Clock: " << tmp << " MHz" << std::endl;
  } else if (clksel == 1) {
    std::cout << "ADCx PHY Clock: " << tmp << " MHz" << std::endl;
  } else if (clksel == 2) {
    std::cout << "LMK Clock: " << tmp << " MHz" << std::endl;
  } else if (clksel == 3) {
    std::cout << "DACx PHY Clock: " << tmp << " MHz" << std::endl;
  } else if (clksel == 4) {
    std::cout << "External Trigger: " << tmp << " MHz" << std::endl;
  }

  return 0;

}

File: Mu2e-STMDAQ/hardware/python/Erdem_source/FMC120_freqcnt.py
#!/usr/bin/python

# -*- coding: utf-8 -*-
import sys
import time
import uhal
import spi_functions as s


def FMC120_freqcnt_getfrequency(wibNode,clksel):
   
   # tell the firmware to start a measure on a given clock index
   wibNode.getNode("axi_fmc120_8lane.Freq_ctr.clk_sel").write(clksel)
   wibNode.dispatch()

   time.sleep(0.5)

   # read back the just measured value
   dword = wibNode.getNode("axi_fmc120_8lane.Freq_ctr.clk_cnt").read()
   wibNode.dispatch()

   cmdfreq = 125.0

   testClkperiod = 1.0 /125
   tmp = 8192 * testClkperiod
   tmp = (tmp /(dword+1))
   tmp = 1.00/tmp

   mult_factor = 4
   
   if (clksel == 0):
       print("Stellar IP Clock: %6.2f MHz" %tmp) 
   elif (clksel == 1):
       print("ADCx PHY Clock: %6.2f MHz" %tmp)
   elif (clksel == 2):
       print("LMK Clock : %6.2f MHz" %tmp)
   elif (clksel == 3):
       print("DACx PHY Clock: %6.2f MHz" %tmp)
   elif (clksel == 4):
       print("External Trigger: %6.2f MHz" %tmp)
   return 0
